aidial_assistant/json_stream/exceptions.py

```diff
@@ -1,10 +1,21 @@
 class JsonParsingException(Exception):
-    pass
+    def __init__(self, message: str, char_position: int):
+        super().__init__(
+            f"Failed to parse json string at position {char_position}: {message}"
+        )
 
 
 def unexpected_symbol_error(
     char: str, char_position: int
 ) -> JsonParsingException:
-    return JsonParsingException(
-        f"Failed to parse json string: unexpected symbol {char} at position {char_position}"
-    )
+    return JsonParsingException(f"Unexpected symbol {char}.", char_position)
+
+
+def unexpected_end_of_stream_error(char_position: int) -> JsonParsingException:
+    return JsonParsingException("Unexpected end of stream.", char_position)
+
+
+def invalid_sequence_error(
+    sequence: str, char_position: int
+) -> JsonParsingException:
+    return JsonParsingException(f"Invalid sequence {sequence}.", char_position)
```

Comment:
What does `sequence` mean? Any string is a sequence of character, so it doesn't help at all.

It think the error message `Can't parse {expected_type} from the string '{string}'` is much more comprehensive.